src/QueryBuilder.php

```diff
@@ -128,6 +127,11 @@ public function __construct(
         $this->parameters = new ArrayCollection();
     }
 
+    public function getType(): QueryType
```

Comment:
Let's not open this too much, shall we?

```suggestion
    final protected function getType(): QueryType
```

Reply:
Should I remove the test then ?

Reply:
This is getting philosophical… I think protected method are part of the public API, so you should keep testing them. You could have

```php
$qb = new class  extends QueryBuilder
{
    public method assertCallingUpdateMakesTheQueryAnUpdate(): void
    {
        $this->update();
        TestCase::assertSame(QueryType::UPDATE, $this->getType());
    }
}

$qb->assertCallingUpdateMakesTheQueryAnUpdate();
```